internal/pkg/deploy/cloudformation/app.go

```diff
@@ -50,7 +52,66 @@ func (cf CloudFormation) DeployApp(in *deploy.CreateAppInput) error {
 }
 
 func (cf CloudFormation) UpgradeApplication(in *deploy.CreateAppInput) error {
-	return nil
+	appConfig := stack.NewAppStackConfig(in)
+	s, err := toStack(appConfig)
+	if err != nil {
+		return err
+	}
+	if err := cf.upgradeAppStack(s); err != nil {
+		return err
+	}
+	return cf.upgradeAppStackSet(appConfig)
+}
+
+func (cf CloudFormation) upgradeAppStackSet(config *stack.AppStackConfig) error {
+	for {
+		ssName := config.StackSetName()
+		if err := cf.appStackSet.WaitForStackSetLastOperationComplete(ssName); err != nil {
+			return fmt.Errorf("wait for stack set %s last operation complete: %w", ssName, err)
+		}
+		previouslyDeployedConfig, err := cf.getLastDeployedAppConfig(config)
+		if err != nil {
+			return err
+		}
+		previouslyDeployedConfig.Version += 1
+		err = cf.deployAppConfig(config, previouslyDeployedConfig)
+		if err == nil {
```

Comment:
Would you mind investigating this new feature: https://aws.amazon.com/about-aws/whats-new/2021/04/deploy-cloudformation-stacks-concurrently-across-multiple-aws-regions-using-aws-cloudformation-stacksets/ that allows stackset instances to be deployed in parallel? 
It would be really cool if we can add that in a follow-up PR so that these upgrades don't take forever